### PR TITLE
Add `createTempFileName` helper.

### PR DESCRIPTION
**Pull request by Fuuzetsu**
See https://github.com/UnkindPartition/temporary/issues/10 and
https://github.com/UnkindPartition/temporary/pull/11 for discussion.

### CODE REVIEW

File: System/IO/Temp.hs [hunk 1]
```diff
@@ -185,6 +186,22 @@ createTempDirectory dir template = findTempName
         Left  e | isAlreadyExistsError e -> findTempName
                 | otherwise              -> ioError e
 
+-- | Get a temporary file name, without creating a file.
+--
+-- This may be useful if you have some process wait for a file creation to
+-- happen.
```
Comment (UnkindPartition) [inline System/IO/Temp.hs]: I don't quite follow what you're getting at here.

To me, "wait for a file creation to happen" would be something like `while ! [ -f "$FILE" ]; do sleep 1; done` — but I don't see how it's related to `createTempFileName`.

Comment (Fuuzetsu) [reply System/IO/Temp.hs]: That is more or less what I wanted to say.

So you can have a process/function/whatever that accepts a filename and then waits for it to appear to do something (via inotify or busy loop or whatever). One could get such a filename via `createTempFileName`.

We can remove this comment if it's confusing, it's probably not useful to try to suggest some use-case.

**Comment by UnkindPartition** [comment]
Thanks! I pushed a commit to your branch expanding the docs and adding a `with` variant. Please take a look and let me know if you're ok with it.

**Comment by Fuuzetsu** [comment]
> Thanks! I pushed a commit to your branch expanding the docs and adding a `with` variant. Please take a look and let me know if you're ok with it.

That's a lot better than what I wrote, thank you. Please merge at will.

**Pull request merged**